### PR TITLE
Add check for organism specific gtf

### PR DESCRIPTION
**Pull request by morsecodist**
# Description
This excludes ERCC.gtf files from counting as organism-specific gtf files when considering collecting insert size metrics for RNA.

# Version
- [x] I have increased the appropriate version number in https://github.com/chanzuckerberg/idseq-dag/blob/master/idseq_dag/__init__.py. Guidelines here: https://github.com/chanzuckerberg/idseq-dag/blob/pr-template/README.md#release-notes
- [x] I have added release notes for my new version to https://github.com/chanzuckerberg/idseq-dag/blob/pr-template/README.md#release-notes

# Tests
- [x] I have verified in IDseq staging that the pipeline still completes successfully:
    - [x] for single-end inputs
    - [x] for paired-end inputs
    - [x] for FASTQ inputs
    - [x] for FASTA inputs.
- [x] I have validated that my change does not introduce any correctness bugs to existing output types.
- [x] I have validated that my change does not introduce significant performance regressions or I have discussed with the team that the benefits of the change are substantial enough that we're comfortable accepting the size of the measured performance penalty.

### CODE REVIEW

File: idseq_dag/steps/run_star.py [hunk 1]
```diff
@@ -116,17 +116,19 @@ def __init__(self, *args, **kwargs):
...
         star_genome_dir = os.path.dirname(self.additional_files.get("star_genome", ""))
-        has_gtf = s3.check_s3_presence_for_pattern(star_genome_dir, r"\.gtf$")
+        has_gtf = s3.check_s3_presence_for_pattern(star_genome_dir, r"(?<!/ERCC)\.gtf$")
 
         self.collect_insert_size_metrics_for = None
         # If we have paired end reads and metrics output files were requested
         #   try to compute insert size metrics
         if paired and requested_insert_size_metrics_output:
-            # Compute for RNA if host genome has a gtf file
+            # Compute for RNA if host genome has an origanisn-specific gtf file
```
Comment (gregdingle) [inline idseq_dag/steps/run_star.py]: sp: organism

File: idseq_dag/steps/run_star.py [hunk 2]
```diff
@@ -116,17 +116,19 @@ def __init__(self, *args, **kwargs):
         self.output_histogram_file = self.additional_attributes.get("output_histogram_file")
         requested_insert_size_metrics_output = bool(self.output_metrics_file or self.output_histogram_file)
 
-        # Check if the STAR genome was generated with a gtf file
+        # Check if the STAR genome was generated with an organism-specific gtf file
         #  This file will be in the same s3 directory, it is needed to prevent overestimation
         #  of insert size for RNA because genomic alignments of RNA may cross introns.
+        #  The ERCC.gtf file is not sufficient for this purpose. An organism-specific gtf
```
Comment (gregdingle) [inline idseq_dag/steps/run_star.py]: When is a ERCC.gtf file generated vs other gtf files. Please add to the comment.

File: idseq_dag/steps/run_star.py [hunk 3]
```diff
@@ -108,27 +108,30 @@ def __init__(self, *args, **kwargs):
         self.sequence_input_files = None
         self.validated_input_counts_file = None
 
+        insert_size_metrics_enabled = kwargs.get("insert_size_metrics_enabled", True)
```
Comment (morsecodist) [inline idseq_dag/steps/run_star.py]: This became an issue during testing. In practice this probably wont happen but I want to be explicit. We never want to run this downstream so I am explicitly disabling it.

File: idseq_dag/steps/run_star.py [hunk 4]
```diff
@@ -103,32 +103,35 @@ class PipelineStepRunStar(PipelineStep):
...
+            # Compute for RNA if host genome has an organism specific gtf file
+            if nucleotide_type == "rna":
+                # Check if the STAR genome was generated with an organism specific gtf file
+                #  This file will be in the same s3 directory, it is needed to prevent overestimation
+                #  of insert size for RNA because genomic alignments of RNA may cross introns.
+                #  The ERCC.gtf file is not sufficient for this purpose. An organism specific gtf
+                #  file will have a name other than ERCC.gtf. The organism specific gtf files
+                #  come from the RefSeq Database.
+                star_genome_dir = os.path.dirname(self.additional_files.get("star_genome", ""))
+                has_gtf = s3.check_s3_presence_for_pattern(star_genome_dir, r"(?<!/ERCC)\.gtf$")
```
Comment (morsecodist) [inline idseq_dag/steps/run_star.py]: This is the relevant regex

Comment (katrinakalantar) [reply idseq_dag/steps/run_star.py]: LGTM

**Pull request merged**